File: 2418-sort-the-people/2418-sort-the-people.java
class Solution {
    public String[] sortPeople(String[] names, int[] heights) {
        HashMap<Integer,String> map = new HashMap<>();
        
        for(int i = 0; i < heights.length; i++) {
            map.put(heights[i],names[i]);
        }
        
        Arrays.sort(heights);
        String[] answer = new String[heights.length];
        int index = heights.length - 1;
        
        for(int height : heights) {
            answer[index] = map.get(height);     
            index--;
        }
        
        return answer;
    }
}

    

File: 2418-sort-the-people/2418-sort-the-people.py
class Solution:
    def sortPeople(self, names: List[str], heights: List[int]) -> List[str]:
        n = len(names)
        
        for i in range(n):
            min_idx = i
            for j in range(i + 1, n):
                if heights[min_idx] < heights[j]:
                    min_idx = j
            names[min_idx], names[i] = names[i], names[min_idx]
            heights[min_idx], heights[i] = heights[i], heights[min_idx]
            
        return names